Tests/SwiftDocCTests/Model/DocumentationMarkupTests.swift

```diff
@@ -158,6 +158,21 @@ class DocumentationMarkupTests: XCTestCase {
             XCTAssertEqual(expected, model.abstractSection?.content.map({ $0.detachedFromParent.debugDescription() }).joined(separator: "\n"))
         }
 
+        // Contains a HTMLBlock before the abstract
+        do {
+            let source = """
+            # Title
+            <!--Line a-->
+            Line b
+            ## Hello, world!
+            Discussion content.
+            """
+            let expected = """
+            Text \"Line b\"
+            """
+            let model = DocumentationMarkup(markup: Document(parsing: source, options: .parseBlockDirectives))
+            XCTAssertEqual(expected, model.abstractSection?.content.map{ $0.detachedFromParent.debugDescription() }.joined(separator: "\n"))
+        }
     }
```

Comment:
Worth adding an explicit test for the `@Comment{}` as well perhaps?

Reply:
Done. You can check and review the change here [5a4d4](https://github.com/apple/swift-docc/pull/347/commits/5a4d41e2bcdb9ef1b24a79853286c64676df63e2)